spring-cloud-dataflow-core/src/main/java/org/springframework/cloud/dataflow/core/TaskDefinition.java

```diff
@@ -38,24 +41,35 @@
 	private final String dslText;
 
 	public TaskDefinition(String name, String dsl) {
+		this(name, dsl, false);
+	}
+
+	TaskDefinition(String registeredAppName, String label, Map<String, String> properties) {
+		super(registeredAppName, label, properties);
+		this.dslText = "";
+		properties.put("spring.cloud.task.name", registeredAppName);
+	}
+
+	public TaskDefinition(String name, String dsl, boolean isComposedTask) {
 		this.dslText = dsl;
-		AppNode taskNode = new TaskParser(name, dsl).parse();
-		setRegisteredAppName(taskNode.getName());
 		Map<String, String> properties = new HashMap<>();
-		if (taskNode.hasArguments()) {
-			for (ArgumentNode argumentNode : taskNode.getArguments()) {
-				properties.put(argumentNode.getName(), argumentNode.getValue());
+		if(isComposedTask) {
```

Comment:
This is never something we particularly talked about really but I can see your point of view. From my point of view, the DSL is complicated enough, if you supported/allowed arguments inside the DSL too everywhere a task could be specified, you are perhaps steering folks further down writing incomprehensible strings (especially as tasks seem to have far more options set). Spring XD didn't do this, but not sure how much that impacts choices made in SCDF.